PetShare/Services/Implementations/Applications/ApplicationCommand.cs

```diff
@@ -85,11 +90,27 @@ public async Task<Result<Application>> AcceptAsync(Guid id, CancellationToken to
             return new InvalidOperation("Application was withdrawn");
         if (entity.State == ApplicationState.Rejected)
             return new InvalidOperation("Application was rejected");
+        if (await _context.Verifications.AnyAsync(e => e.AdopterId == id && e.ShelterId == entity.Announcement.AuthorId, token))
+            return new InvalidOperation("Adopter is not verified");
 
         entity.State = ApplicationState.Accepted;
         entity.LastUpdateTime = DateTime.Now;
+        entity.Announcement.Status = AnnouncementStatus.Closed;
+        entity.Announcement.LastUpdateDate = DateTime.Now;
+        entity.Announcement.ClosingDate = DateTime.Now;
         await _context.SaveChangesAsync(token);
 
+        await _emailService.SendStatusUpdateEmail(entity.Adopter.Email, entity.Adopter.UserName, ApplicationState.Accepted.ToString());
+
+        var otherApplications = await _context.Applications.
+            Include(app => app.Announcement).
+            Where(app => app.Announcement.Id == entity.AnnouncementId && app.State == ApplicationState.Created).
+            ToListAsync();
+        foreach (var app in otherApplications)
+        {
+            await RejectAsync(app.Id, token);
+        }
```

Comment:
We should do all these db operations in one go, save them once, and then send relevant emails if the operation was successful. Right now if anyone does anything with any of the applications that are being rejected while that happens, we will have an invalid state in db. 

It's not _really_ crucial in our case, but it's a simple fix so maybe let's do that.